src/main/java/org/broadinstitute/hellbender/engine/FeatureInput.java

```diff
@@ -70,6 +87,123 @@
      */
     public static final String FEATURE_ARGUMENT_KEY_VALUE_SEPARATOR = "=";
 
+    public static boolean isGenomeDBPath(String filePath){
+        return filePath.startsWith(GENOMIC_DB_URI_SCHEME);
+    }
+
+    @SuppressWarnings("unchecked")
+    private FeatureCodec<T, ?> getCodec() {
+        if( codec != null){
+            return codec;
+        }
+
+        if( featureType == null){
+            throw new GATKException("setFeatureType must be called before calling getCodec");
+        }
+        codec = (FeatureCodec<T, ?>) FeatureManager.getCodecForFile(new File(getFeaturePath()), featureType);
+
+        return codec;
+    }
+
+    @SuppressWarnings("unchecked")
+    public FeatureReader<T> getFeatureReader() {
```

Comment:
@kgururaj That may be, but the GATK engine needs to work with both, so we have to enforce this limitation.